pkg/ipamd/ipamd.go

```diff
@@ -915,6 +915,8 @@ func (c *IPAMContext) tryAssignCidrs() (increasedPool bool, err error) {
 // For an ENI, try to fill in missing IPs on an existing ENI.
 // PRECONDITION: isDatastorePoolTooLow returned true
 func (c *IPAMContext) tryAssignIPs() (increasedPool bool, err error) {
+	insufficientENI := make(map[*datastore.ENI]bool)
```

Comment:
Each time we call `tryAssignIps` or `tryAssignPrefixes` we create a new map.. `c.dataStore.GetENINeedsIP(c.maxIPsPerENI, c.useCustomNetworking, c.useSubnetDiscovery, insufficientENI)` so here, we are always passing new map isn't it or am I missing something?